ObjectiveCloudant/Operations/CDTQueryViewOperation.m

```diff
@@ -0,0 +1,295 @@
+    }
+
+    NSURLQueryItem *stale = [CDTQueryViewOperation QueryItemForStaleState:self.stale];
+
+    if (stale) {
+        [queryItems addObject:stale];
+    }
+
+    if (self.startKey) {
+        [queryItems
+            addObject:[NSURLQueryItem
+                          queryItemWithName:@"startkey"
+                                      value:[CDTQueryViewOperation stringForObject:self.startKey]]];
+    }
+
+    if (self.startKeyDocId) {
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:@"startkey_docid"
+                                                          value:self.startKeyDocId]];
+    }
+
+    return [queryItems copy];
+}
+
+- (void)callCompletionHandlerWithError:(NSError *)error
+{
+    if (self && self.queryViewCompletionBlock) {
+        self.queryViewCompletionBlock(error);
+    }
+}
+
+- (void)processResponseWithData:(NSData *)responseData
+                     statusCode:(NSInteger)statusCode
+                          error:(NSError *)error
+{
+    if (error) {
+        [self callCompletionHandlerWithError:error];
+    } else if (statusCode / 100 == 2) {
+        // process the things!
+        NSError *jsonError;
+        NSDictionary<NSString *, NSObject *> *responseJson =
+            [NSJSONSerialization JSONObjectWithData:responseData options:0 error:&jsonError];
+
+        if (!responseJson) {
+            [self callCompletionHandlerWithError:jsonError];
+        } else {
+            if (self.viewRowBlock) {
+                for (NSDictionary<NSString *, NSObject *> *row in(
+                         NSArray<NSDictionary<NSString *, NSObject *> *> *)responseJson[@"rows"]) {
+                    self.viewRowBlock(row);
+                }
+            }
+            if (self.queryViewCompletionBlock) {
+                self.queryViewCompletionBlock(nil);
+            }
+        }
+
+    } else {
+        NSString *json = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
+        NSString *msg = [NSString
+            stringWithFormat:@"Querying view failed with %ld %@.", (long)statusCode, json];
+        NSDictionary *userInfo = @{NSLocalizedDescriptionKey : NSLocalizedString(msg, nil)};
+        error = [NSError errorWithDomain:CDTObjectiveCloudantErrorDomain
+                                    code:CDTObjectiveCloudantErrorQueryViewFailed
+                                userInfo:userInfo];
+
+        [self callCompletionHandlerWithError:error];
+    }
+}
+
++ (BOOL)isValidStringOrJsonArray:(NSObject *)obj
+{
+    if ([obj isKindOfClass:[NSArray class]]) {
+        if (![NSJSONSerialization isValidJSONObject:obj]) {
+            return NO;
+        }
+    } else if (![obj isKindOfClass:[NSString class]]) {
+        return NO;
+    }
+
+    return YES;
+}
+
++ (NSString *)stringForBoolean:(BOOL)boolean
+{
+    if (boolean) {
+        return @"true";
+    } else {
+        return @"false";
+    }
+}
```

Comment:
I would name these "stringForX" methods something more like "queryStringRepresentationForX" to make it more obvious what they do. Otherwise the name isn't descriptive of what comes out of the method, at least more than "you'll get something random that's an NSString".